react-native-async-storage/async-storage
# chore: bump react-native to 0.71

## Summary

This PR also removes Detox. Detox keeps breaking with every update and they're really hard to figure out every time. Tests are disabled until we can implement an alternative.

- `react-native` ⇧ 0.71.11
- `react-native-macos` ⇧ 0.71.13
- `react-native-web` ⇧ 0.18.12
- `react-native-windows` ⇧ 0.71.18
- `expo` ⇧ 48.0.11
- `detox` ✘ _removed_

## Test Plan

- Make sure all the test apps work
  - Android
  - iOS
  - macOS
  - Windows
- Verify that Expo/web works

## Screenshots

| Android | iOS | macOS | Windows |
| :-: | :-: | :-: | :-: |
| ![Screenshot_1681477733](https://user-images.githubusercontent.com/4123478/232052547-02d5ae66-1ac7-4d03-b1cf-753d12d97214.png) | ![Simulator Screen Shot - iPhone 14 - 2023-04-14 at 15 01 33](https://user-images.githubusercontent.com/4123478/232052590-8e1614ff-c874-4b50-9048-4341461b8262.png) | ![image](https://user-images.githubusercontent.com/4123478/232055174-802ff391-3a74-490e-9dc0-74441a40221b.png) | TODO |
